cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/CopyNativeLibTask.java

```diff
@@ -239,6 +244,46 @@ private void copyLibsFromBalo(Path baloFilePath, Set<Path> moduleDependencySet)
         }
     }
 
+    private void copyAndValidateBaloDependencies(BuildContext buildContext, Path importDependencyPath,
+                                                 PackageID packageID, Path project, Set<Path> moduleDependencySet) {
+        Manifest manifestFromBalo = RepoUtils.getManifestFromBalo(importDependencyPath);
+        List<Library> baloDependencies = manifestFromBalo.getPlatform().libraries;
+        List<Library> libraries = manifest.getPlatform().libraries;
+        HashSet<Path> baloCompileScopeDependencies = new HashSet<>();
+        HashSet<Path> platformLibs = new HashSet<>();
+
+        if (baloDependencies == null) {
+            return;
+        }
+
+        for (Library baloTomlLib : baloDependencies) {
+            if (baloTomlLib.getScope() != null && baloTomlLib.getScope().equalsIgnoreCase("provided")) {
+                baloCompileScopeDependencies.add(Paths.get(baloTomlLib.getPath()).getFileName());
+            }
+        }
+
+        if (libraries != null) {
+            for (Library library : libraries) {
+                if (Arrays.asList(library.getModules()).contains(packageID.orgName.value + "/" +
+                        packageID.name.value)) {
+                    Path libFilePath = Paths.get(library.getPath());
+                    platformLibs.add(libFilePath.getFileName());
+                    moduleDependencySet.add(project.resolve(libFilePath));
+                }
+            }
+        }
+
+        if (platformLibs.containsAll(baloCompileScopeDependencies)) {
+            return;
+        }
+
+        baloCompileScopeDependencies.removeAll(platformLibs);
+        for (Path baloTomlLib : baloCompileScopeDependencies) {
+            buildContext.out().println("warning: " + packageID + " is missing a native library dependency - " +
```

Comment:
can we check !platformLibs.contains(baloTomlLib) and then log warning. so we can get rid of 276 to 280 lines